js/ai/src/resource.ts

```diff
@@ -29,6 +29,11 @@ import { Part, PartSchema } from './model.js';
  * Options for defining a resource.
  */
 export interface ResourceOptions {
+  /**
+   * Resource name.
+   */
+  name: string;
```

Comment:
let's make this optional with the name defaulting to the specified uri or template

Reply:
Done.